jablib/src/main/java/org/jabref/logic/importer/util/FileFieldParser.java

```diff
@@ -46,11 +49,16 @@ public FileFieldParser(String value) {
     public static List<LinkedFile> parse(String value) {
         // We need state to have a more clean code. Thus, we instantiate the class and then return the result
         FileFieldParser fileFieldParser = new FileFieldParser(value);
-        return fileFieldParser.parse();
+        return new ArrayList<>(fileFieldParser.parse().stream().map(LinkedFilePosition::linkedFile).toList());
```

Comment:
To create a modifiable list, the usual way is with `.collect(Colltectors.toList())` (or similar).

Also add to JavaDoc that the list is modifiable.

However - why do you need the result be modifiable? I think, the caller should convert it to a modifiable list (I might be wrong here, did not check all the code, but only thought of parsing and that the parsed representation should be read-only, because the underlying thing does not change)